### ISSUE TITLE
[Status token colors] improve contrast between text on dark mode danger/brand colors

### ISSUE THREAD

**Issue by SkyeSeitz** [Esri/calcite-components#13234]
**monday.com sync:** #18314238467

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

N/A

### Actual Behavior

Currently, the semantic `status.brand.hover`/`press` and `status.danger.hover`/`press` colors do not have sufficient contrast with text colors in dark mode. Ideally, these would be compatible with `text.inverse`

<img width="640" alt="Image" src="https://github.com/user-attachments/assets/ee4c0254-9a0d-4b17-8b6a-15a5cc96a805" />

Since significant value changes are expected for the previously mentioned colors, the remaining semantic status colors in dark mode should be updated to follow a similar progression and maintain consistency

<img width="640" alt="Image" src="https://github.com/user-attachments/assets/f82b8f59-a4d8-4364-b946-3f463e3b97c8" />

### Expected Behavior

Colors remain accessible in hover and press states for all brand and status tokens. [See Figma file for full specs 🔗](https://www.figma.com/design/Kfqw5yJBwTRrvUsm20NUJt/-Status-token-colors--improve-contrast-between-text-on-dark-mode-status-colors?node-id=3848-11472&t=Ve1H19eQNdtgdeJx-11)

#### Semantic updates

updates to semantic.color.brand (dark) tokens:
- [x] `brand.hover` to reference `core.color.dark.blue.d-bb-410`
- [x] `brand.press`. to reference `core.color.vibrant.blue.v-bb-110`

updates to semantic.color.info (dark) tokens:
- [x] `info.hover` to reference `core.color.dark.blue.d-bb-410`
- [x] `info.press` to reference `core.color.vibrant.blue.v-bb-110`

updates to semantic.color.success (dark) tokens:
- [x] `success.hover` to reference `core.color.vibrant.green.v-gg-120`
- [x] `success.press` to reference `core.color.vibrant.green.v-gg-110`

updates to semantic.color.warning (dark) tokens:
- [x] `warning.hover` to reference `core.color.high-saturation.orange-yellow.h-oy-050`
- [x] `warning.press` to reference `core.color.high-saturation.orange-yellow.h-oy-030`

updates to semantic.color.danger(dark) tokens:
- [x] `danger.hover` to reference `core.color.dark.red.d-rr-410`
- [x] `danger.press` to reference `core.color.vibrant.red.v-rr-110`

<img width="2096" alt="Image" src="https://github.com/user-attachments/assets/f6aeae59-0fea-4c24-b8a3-298c063ad91e" />

#### Core token updates

- [x] update `core.color.high-saturation.orange-yellow.h-oy-030`’s value to `#FFC47D`
- [x] update `core.color.high-saturation.orange-yellow.h-oy-050`’s value to `#FFAF4F`
- [x] update `core.color.vibrant.green.v-gg-120`’s value to `#6DF278`
- [x] update `core.color.dark.blue.d-bb-410`’s value to `#40B9FF`
- [x] update `core.color.dark.red.d-rr-410`’s value to `#FE7863`

<img width="2080" alt="Image" src="https://github.com/user-attachments/assets/5855397b-4358-4798-8fde-95e5bbb505f9" />

#### Core token additions

- [x] add `core.color.vibrant.blue.v-bb-110` with a value of `#80D1FF`
- [x] add `core.color.vibrant.green.v-gg-110` with a value of `#9DF2A3`
- [x] add `core.color.vibrant.red.v-rr-110` with a value of `#FE9989`

<img height="200" alt="Image" src="https://github.com/user-attachments/assets/969f07be-6819-4d82-a5ec-ac91c17d1b11" />

### Reproduction Sample

https://codepen.io/matgalla/pen/PwzXORG

### Reproduction Steps

1. Set Buttons to hover and press states
2. Inspect color contrast with text

### Reproduction Version

5.0.1

### Relevant Info

Blocked by #13233 

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [x] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by DintaMel** [comment]
Verified in Codepen using Calcite 5.0.3-next.2 and Calcite design tokens 4.1.0-next.1.

[Semantic token changes codepen ](https://codepen.io/DintaMel-the-lessful/pen/raMxxQP?editors=1000)
[Core token changes codepen](https://codepen.io/DintaMel-the-lessful/pen/xbEZOKd?editors=1000)
